api/v1alpha1/cachefsruntime_types.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2024 The Fluid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	CacheFSRuntimeKind = "CacheFSRuntime"
+)
+
+// CacheFSRuntimeSpec defines the desired state of CacheFSRuntime
+type CacheFSRuntimeSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// The version information that instructs fluid to orchestrate a particular version of CacheFS.
+	CacheFSVersion VersionSpec `json:"cachefsVersion,omitempty"`
+
+	// The component spec of CacheFS worker
+	Worker CacheFSCompSpec `json:"worker,omitempty"`
+
+	// Desired state for CacheFS Fuse
+	Fuse CacheFSFuseSpec `json:"fuse,omitempty"`
+
+	// Tiered storage used by CacheFS
+	TieredStore TieredStore `json:"tieredstore,omitempty"`
+
+	// Dataset used as CacheFS source with optional subdir, e.g. "dataset_name[:/subdir]"
+	ImportDataset string `json:"importDataset,omitempty"`
+
+	// Configs of CacheFS, options for cachefs format sub-command
+	Configs map[string]string `json:"configs,omitempty"`
+
+	// The replicas of the worker, need to be specified
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// Manage the user to run Cachefs Runtime
+	RunAs *User `json:"runAs,omitempty"`
+
+	// Volumes is the list of Kubernetes volumes that can be mounted by the cachefs runtime components and/or fuses.
+	// +optional
+	Volumes []corev1.Volume `json:"volumes,omitempty"`
+
+	// PodMetadata defines labels and annotations that will be propagated to CacheFs's pods.
+	// +optional
+	PodMetadata PodMetadata `json:"podMetadata,omitempty"`
+}
+
+// CacheFSCompSpec is a description of the CacheFS components
+type CacheFSCompSpec struct {
+	// Replicas is the desired number of replicas of the given template.
+	// If unspecified, defaults to 1.
+	// +kubebuilder:validation:Minimum=1
```

Comment:
If not specified, the replicas should be 0. This follows Kubernetes' convention.

Reply:
And why do we need set the replicas inside CacheFSCompSpec?

Reply:
Seems duplicated with CacheFSRuntime.Replicas, will remove.

Reply:
Fixed